services/validatorWorker/leader.js

```diff
@@ -1,14 +1,24 @@
 const { persistAndPropagate } = require('./lib/propagation')
-const { getStateRootHash } = require('./lib/followerRules')
-
+const { getStateRootHash } = require('./lib')
 const producer = require('./producer')
+const { heartbeat } = require('./heartbeat')
 
 function tick(adapter, channel) {
 	return producer.tick(channel)
 		.then(
 			res => res.newStateTree ?
 				afterProducer(adapter, res)
 				: { nothingNew: true }
+		).then(
+			res => {
+				if(res && res.nothingNew){
+					// send heartbeat
+					Promise.resolve(
```

Comment:
yeah, the heartbeat is returning a promise, so this should just be `return heartbeat(adapter, channel)`